bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// Rke2ConfigSpec defines the desired state of Rke2Config
-type Rke2ConfigSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// RKE2ConfigSpec defines the desired state of RKE2Config
+type RKE2ConfigSpec struct {
+	// Files specifies extra files to be passed to user_data upon creation.
+	// +optional
+	Files []File `json:"files,omitempty"`
 
-	// Foo is an example field of Rke2Config. Edit rke2config_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// PreRKE2Commands specifies extra commands to run before rke2 setup runs
+	// +optional
+	PreRKE2Commands []string `json:"preRKE2Commands,omitempty"`
+
+	// DeployRKE2Commands enables to override commands that are executed to setup rke2.
+	// It may be leveraged to deploy
+	// +optional
+	DeployRKE2Commands []string `json:"deployRKE2Commands,omitempty"`
+
+	// PostRKE2Commands specifies extra commands to run after rke2 setup runs
+	// +optional
+	PostRKE2Commands []string `json:"postRKE2Commands,omitempty"`
+
+	// AgentConfig specifies configuration for the agent nodes
+	// +optional
+	AgentConfig RKE2AgentConfig `json:"agentConfig,omitempty"`
+
+	// ServerConfig specifies configuration for the agent nodes
+	// +optional
+	ServerConfig RKE2ServerConfig `json:"serverConfig,omitempty"`
+
+	// PrivateRegistriesConfig defines the containerd configuration for private registries and local registry mirrors.
+	//+optional
+	PrivateRegistriesConfig Registry `json:"privateRegistriesConfig,omitempty"`
+
+	// Version specifies the rke2 version
+	// +optional
+	Version string `json:"version,omitempty"`
 }
 
-// Rke2ConfigStatus defines the observed state of Rke2Config
-type Rke2ConfigStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+type RKE2ServerConfig struct {
+
+	// BindAddress describes the rke2 bind address (default: 0.0.0.0)
+	// +optional
+	BindAddress string `json:"bindAddress,omitempty"`
+
+	// AdvertiseAddress IP address that apiserver uses to advertise to members of the cluster (default: node-external-ip/node-ip)
+	// +optional
+	AdvertiseAddress string `json:"advertiseAddress,omitempty"`
+
+	// TLSSan Add additional hostname or IP as a Subject Alternative Name in the TLS cert
+	// +optional
+	TLSSan []string `json:"tlsSan,omitempty"`
+
+	// DataDir is the Folder to hold RKE2's state (default: "/var/lib/rancher/rke2")
+	//+optional
+	DataDir string `json:"data-dir,omitempty"`
+
+	// ServiceNodePortRange is the port range to reserve for services with NodePort visibility (default: "30000-32767")
+	//+optional
+	ServiceNodePortRange string `json:"service-node-port-range,omitempty"`
```

Comment:
Well, in that case, it would need to be 2 ints, one for the `Range Max` and one for the `Range Min`. But, the idea is not bad, it will need however some research on what are the possible values.

Reply:
Oh, it's a range, I didn't notice. Let's keep the string for simplicity.

Reply:
OK, I will resolve this conversation then!